xapian-core/cluster/cluster.cc

```diff
@@ -296,9 +296,9 @@ Point::Point(const TermListGroup &tlg, const Document &document_)
 	string term = *it;
 	double termfreq = tlg.get_termfreq(term);
 
-	/// If term frequency of the term is lesser than or equal
-	/// to 1 or wdf is lesser than 1, we do not consider this
-	/// term for document vector calculations
+	// If the term doesn't exist within the document or
+	// exists in only one document within the MSet, then the
+	// terms are not used for document vector calculations
```

Comment:
"If the term doesn't exist within the document" isn't what `wdf < 1` is checking.  We got this term by iterating the document's termlist, so the term clearly does exist within the document.

I'm not quite sure what this should say though.

The wdf is unsigned, so the only possible `wdf < 1` case is when it's zero - that's true for filter terms (terms added with `add_boolean_term()`), but filter terms are potentially of use for clustering.  Possibly `wdf < 1` isn't a valid reason to reject a term and we should just clamp the wdf to be >= 1 if 0 is a problematic value for the clustering (that's what the weighting scheme calculations do if such a term is used such that it gets weighted).